### PR TITLE
Apply some ReSharper fixes to our code

### PR DESCRIPTION
**Pull request by pleroy**
Two types of changes:
1. Systematic changes of things that we always want (e.g., unused `using` clauses).
1. Proposed changes that we may or may not want applied by hand.

No contentious things like adjusting naming conventions or reformatting.

### CODE REVIEW

File: ksp_plugin_adapter/main_window.cs [hunk 1]
```diff
@@ -367,13 +364,13 @@ internal class MainWindow : SupervisedWindowRenderer {
           value   : must_record_journal_,
           text    : "Record journal (starts on load)");
       UnityEngine.GUILayout.Label(
-          "Journalling is " + (journaling_ ? "ON" : "OFF"),
+          "Journalling is " + (journalling_ ? "ON" : "OFF"),
```
Comment (eggrobin) [inline ksp_plugin_adapter/main_window.cs]: It seems the spelling of the variable was good and that the string should be changed instead: https://www.lexico.com/en/definition/journal.

File: ksp_plugin_adapter/optional_marshaler.cs [hunk 2]
```diff
@@ -35,16 +35,14 @@ internal class OptionalMarshaler<T> : ICustomMarshaler where T : struct {
       throw Log.Fatal("The runtime returns null for null objects");
     }
     T value;
-    var value_if_boxed = managed_object as T?;
-    if (value_if_boxed != null) {
-      value = value_if_boxed.Value;
+    if (managed_object is T value_if_boxed) {
```
Comment (eggrobin) [inline ksp_plugin_adapter/optional_marshaler.cs]: Fancy!

File: ksp_plugin_adapter/main_window.cs (hunk 1 above)
Comment (pleroy) [reply ksp_plugin_adapter/main_window.cs]: Done.

File: ksp_plugin_adapter/optional_marshaler.cs (hunk 2 above)
Comment (pleroy) [reply ksp_plugin_adapter/optional_marshaler.cs]: Indeed, it does that with a single click.  I didn't even know that this existed.

**Pull request merged**